src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java

```diff
@@ -15,47 +20,195 @@
+            }
+        }
+
+        @Override
+        public void onLocationUpdated(Context context, Location location, RadarUser user) {
+            if (RadarCordovaPlugin.locationCallbackContext == null) {
+                return;
+            }
+
+            try {
+                JSONObject obj = new JSONObject();
+                obj.put("location", Radar.jsonForLocation(location));
+                obj.put("user", user.toJson());
+
+                PluginResult pluginResult = new PluginResult(PluginResult.Status.OK, obj);
+                pluginResult.setKeepCallback(true);
+                RadarCordovaPlugin.locationCallbackContext.sendPluginResult(pluginResult);
+            } catch (JSONException e) {
+                Log.e("RadarCordovaPlugin", "JSONException", e);
+                RadarCordovaPlugin.locationCallbackContext.sendPluginResult(new PluginResult(PluginResult.Status.JSON_EXCEPTION));
+            }
+        }
+
+        @Override
+        public void onClientLocationUpdated(Context context, Location location, boolean stopped, Radar.RadarLocationSource source) {
+            if (RadarCordovaPlugin.clientLocationCallbackContext == null) {
+                return;
+            }
+
+            try {
+                JSONObject obj = new JSONObject();
+                obj.put("location", Radar.jsonForLocation(location));
+                obj.put("stopped", stopped);
+                obj.put("source", source.toString());
+
+                PluginResult pluginResult = new PluginResult(PluginResult.Status.OK, obj);
+                pluginResult.setKeepCallback(true);
+                RadarCordovaPlugin.clientLocationCallbackContext.sendPluginResult(pluginResult);
+            } catch (JSONException e) {
+                Log.e("RadarCordovaPlugin", "JSONException", e);
+                RadarCordovaPlugin.clientLocationCallbackContext.sendPluginResult(new PluginResult(PluginResult.Status.JSON_EXCEPTION));
+            }
+        }
+
+        @Override
+        public void onError(Context context, Radar.RadarStatus status) {
+            if (RadarCordovaPlugin.errorCallbackContext == null) {
+                return;
+            }
+
+            try {
+                JSONObject obj = new JSONObject();
+                obj.put("status", status.toString());
+
+                PluginResult pluginResult = new PluginResult(PluginResult.Status.OK, obj);
+                pluginResult.setKeepCallback(true);
+                RadarCordovaPlugin.errorCallbackContext.sendPluginResult(pluginResult);
+            } catch (JSONException e) {
+                Log.e("RadarCordovaPlugin", "JSONException", e);
+                RadarCordovaPlugin.errorCallbackContext.sendPluginResult(new PluginResult(PluginResult.Status.JSON_EXCEPTION));
+            }
+        }
+
+        @Override
+        public void onLog(Context context, String message) {
+
```

Comment:
should we log here as well? or would that be redundant to the log that triggered this receiver?

Reply:
This is the core SDK logging that comes from `Radar.setLogLevel()`. I actually haven't exposed that function in this lib, just added this stub so all the methods from the abstract `RadarReceiver` class are implemented.

Reply:
gotcha, I guess it'd be redundant to log here then, fine to leave as a noop